### PR TITLE
Switch to Dist::Zilla

### PR DESCRIPTION
**Pull request by davel**

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -43,8 +43,7 @@ jobs:
         with:
           perl-version: ${{ matrix.perl }}
       - run: perl -V
-      - run: cpanm --quiet --notest --installdeps .
-      - run: cpanm --quiet --notest Test::Pod Test::Spelling Test::Perl::Critic
+      - run: cpanm --quiet --notest --installdeps --with-recommends .
       - run: perl Makefile.PL
```
Comment (abraxxa) [inline .github/workflows/actions.yml:47]: This should run dzil test or something similar as there is no Makefile.PL any more.

File: dist.ini [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+[Prereqs / TestRequires]
+Test::More = 0
+Test::Exception = 0
+version = 0
+
+[Prereqs / TestRecommends]
+Test::Spelling = 0
+Test::Perl::Critic = 0
+Test::Pod = 0
```
Comment (abraxxa) [inline dist.ini:44]: I suggest to use various Dist::Zilla::Plugin::Test plugins which add their requirements and also generate the test files.

File: dist.ini [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+x_contributors = Nicolas R <atoomic@cpan.org>
+x_contributors = Dave Lambley <dlambley@cpan.org>
+x_contributors = Ruslan Zakirov <ruz@bestpractical.com>
+x_contributors = Masahito Ikuta <cooldaemon@gmail.com>
+x_contributors = Rod Taylor <rod.taylor@gmail.com>
+x_contributors = Carl Hörberg <carl@cloudamqp.com>
+
+[InstallGuide]
+;[CPANFile]
+[VersionFromModule]
```
Comment (abraxxa) [inline dist.ini:21]: Dist::Zilla does this by default, no need to a plugin.

File: dist.ini [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+[MetaJSON]
+
+[Repository]
+repository = git://github.com/bobtfish/AnyEvent-RabbitMQ.git
```
Comment (abraxxa) [inline dist.ini:25]: https://metacpan.org/pod/Dist::Zilla::Plugin::MetaResources is core Dist::Zilla and has more options, better than another author dependency.

File: dist.ini [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+[Prereqs]
+List::MoreUtils = 0
+Net::AMQP = 0.06
+AnyEvent = 0
+Devel::GlobalDestruction = 0
+namespace::clean = 0
+File::ShareDir = 0
+Readonly = 1.03
```
Comment (abraxxa) [inline dist.ini:34]: The Perl 5.010 requirement was lost.

File: .github/workflows/actions.yml (hunk 1 above)
Comment (davel) [reply .github/workflows/actions.yml:47]: Thanks for taking a look @abraxxa! The integration test is done against the built tarball. By this stage, we have a Makefile.PL, and anything else dzil generates.

Comment (abraxxa) [reply .github/workflows/actions.yml:47]: I see, thanks for the clarification!

File: dist.ini (hunk 3 above)
Comment (davel) [reply dist.ini:21]: That doesn't seem to work on my (Ubuntu 18.04) machine,
```
$ dzil build
[DZ] beginning to build AnyEvent-RabbitMQ
[DZ] no version was ever set
[DZ] no version was ever set at inline delegation in Dist::Zilla for logger->log_fatal (attribute declared in /usr/share/perl5/Dist/Zilla.pm at line 766) line 18.
```

Comment (abraxxa) [reply dist.ini:21]: Sorry, I mixed it up! I'm using PkgVersion so I only need to set the version in dist.ini and it gets added to each module.

Comment (davel) [reply dist.ini:21]: I've not used PkgVersion, but patches welcome of course!

File: xt/03_pod.t [hunk 6]
```diff
@@ -1,3 +1,6 @@
+# XXX should be replaced by Dist::Zilla::Plugin::PodSyntaxTests, but holding
```
Comment (abraxxa) [inline xt/03_pod.t:1]: What's wrong with Dist::Zilla::Plugin::PodSyntaxTests on Ubuntu 18.04?

Comment (davel) [reply xt/03_pod.t:1]: Unlike the current dependencies, it's not packaged, so it's a less convenient install.

I have spotted another problem, perhaps you can help? The META.json contains more data than META.yml, for example under `repository`. Any idea why?

Comment (abraxxa) [reply xt/03_pod.t]: Packaged as in 'deb package'? I've never used distro Perl module packages as I can't chose the version or update it if needed.

Regarding META.json: I *think* that version 2 of the spec is the JSON file and the YAML is for backward compat, see https://metacpan.org/pod/CPAN::Meta::Spec#SERIALIZATION

Comment (davel) [reply xt/03_pod.t]: Thanks! I was worried there was an order problem in `dist.ini`.

Yes, deb packages. I've worked at a few places which have used the distribution's Perl and libraries. It has its pros and cons.

I'll merge this soon if you're okay with that?

Comment (abraxxa) [reply xt/03_pod.t]: Sure, thanks for your work!

**Pull request merged**